presto-array/src/main/java/com/facebook/presto/array/ReferenceCountMap.java

```diff
@@ -59,19 +62,8 @@ public long sizeOf()
         return INSTANCE_SIZE + SizeOf.sizeOf(key) + SizeOf.sizeOf(value) + SizeOf.sizeOf(used);
     }
 
-    private static final class ObjectStrategy
-            implements Strategy<Object>
+    private static long getHashCode(Object key, int size)
     {
-        @Override
-        public int hashCode(Object object)
-        {
-            return System.identityHashCode(object);
-        }
-
-        @Override
-        public boolean equals(Object left, Object right)
-        {
-            return left == right;
-        }
+        return (((long) System.identityHashCode(key)) << Integer.SIZE) + size;
```

Comment:
just curious: woudn't `System.identityHashCode` alone be enough?

Reply:
`System.identityHashCode` has collisions (even when all the objects are still alive). (https://bugs.java.com/bugdatabase/view_bug.do?bug_id=6321873). From benchmark with 100 million objects, 2.5% of them can have collisions.